docs/architecture/blazor-for-web-forms-developers/introduction.md

```diff
@@ -8,6 +8,50 @@ ms.date: 09/11/2019
+
+However, browser have become incredible platforms implementing an ever increasing number of open web standards that grant access to the capabilities of the user's machine. Why not take advantage of the compute power, storage, memory, and other resources of the client device? UI interactions in particular can benefit from a much more interactive feel when handled at least partially or completely client-side. Logic and data that should be handled on the server can still be handled server-side using Web API calls or even over real-time protocols, like WebSockets. All of these benefits are available to web developers for free as long as they are willing to write some JavaScript. Client-side UI frameworks that simplify browser development, like Angular, React, and Vue, have exploded in popularity. ASP.NET Web Forms developers can also benefit from leveraging, and even have some out of the box support with integrated JavaScript frameworks like ASP.NET AJAX.
+
+But having to bridge two different platforms and ecosystems (.NET and JavaScript) comes with a cost. You as the web developer have to become an expert in two parallel worlds with different languages, frameworks, and tools. Code and logic cannot be easily shared between client and server resulting in duplication and engineering overhead. For some, it is also difficult to keep up with the JavaScript ecosystem, which has a history of evolving at breakneck speed, quickly changing preferences for different frontend frameworks (jQuery, Knockout, Angular, React, Vue, etc.) and build tools (grunt, gulp, webpack, etc.). But given JavaScript's browser monopoly there was little choice in the matter. That is until the web community got together and caused a *miracle* to happen!
+
+## WebAssembly to the rescue!
+
+In 2015 the major browser vendors joined forces in a W3C Community Group to create a new open web standard called WebAssembly. WebAssembly is a byte code for the Web. The idea is that if you can compile your code to WebAssembly it can then run on any browser on any platform at near native speed. Initial efforts focused on C/C++ resulting in some pretty dramatic demonstrations of getting native 3D graphics engines running directly in the browser without any plugins. WebAssembly has since been standardized and implemented by all major browsers. Work on running .NET on WebAssembly was announced in late 2017 and is expected to ship in 2020 including support from .NET 5. The ability to run .NET code directly in the browser enables full-stack web development with .NET.
```

Comment:
```suggestion
In 2015 the major browser vendors joined forces in a W3C Community Group to create a new open web standard called WebAssembly. WebAssembly is a byte code for the Web. The idea is that if you can compile your code to WebAssembly, it can then run on any browser on any platform at near native speed. Initial efforts focused on C/C++, resulting in some pretty dramatic demonstrations of getting native 3D graphics engines running directly in the browser without any plugins. WebAssembly has since been standardized and implemented by all major browsers. Work on running .NET on WebAssembly was announced in late 2017 and is expected to ship in 2020, including support from .NET 5. The ability to run .NET code directly in the browser enables full-stack web development with .NET.
```